src/model.rs

```diff
@@ -760,6 +760,70 @@ impl Collections {
         self.vehicle_journeys = CollectionWithId::new(vehicle_journeys).unwrap();
     }
 
+    /// Some comments are identical and can be deduplicated
+    pub fn comment_deduplication(&mut self) {
+        let duplicate2ref = &self.get_comment_map_duplicate_to_referent();
+        if duplicate2ref.is_empty() {
+            return;
+        }
+
+        replace_comment_duplicates_by_ref(&mut self.lines, &duplicate2ref);
+        replace_comment_duplicates_by_ref(&mut self.routes, &duplicate2ref);
+        replace_comment_duplicates_by_ref(&mut self.stop_areas, &duplicate2ref);
+        replace_comment_duplicates_by_ref(&mut self.stop_points, &duplicate2ref);
+        replace_comment_duplicates_by_ref(&mut self.stop_locations, &duplicate2ref);
+
+        fn replace_comment_duplicates_by_ref<T>(
+            collection: &mut CollectionWithId<T>,
+            duplicate2ref: &BTreeMap<String, String>,
+        ) where
+            T: Id<T> + CommentLinks,
+        {
+            let mut map_pt_object_duplicates: BTreeMap<Idx<T>, Vec<&str>> = BTreeMap::new();
+            for (idx, pt_object) in collection.iter() {
+                let mut intersection = vec![];
+                for comment_id in pt_object.comment_links() {
+                    if let Some((duplicate_id_ref, _)) = duplicate2ref.get_key_value(comment_id) {
+                        intersection.push(duplicate_id_ref.as_str());
+                    }
+                }
+                if !intersection.is_empty() {
+                    map_pt_object_duplicates.insert(idx, intersection);
+                }
+            }
```

Comment:
if you want "rustic"  code

```suggestion
           let map_pt_object_duplicates: BTreeMap<Idx<T>, Vec<&str>> = collection
                .iter()
                .filter_map(|(idx, pt_object)| {
                    let intersection: Vec<&str> =
                        pt_object.comment_links().iter().filter_map(|comment_id| {
                            duplicate2ref
                                .get_key_value(comment_id)
                                .map(|(duplicate_id_ref, _)| duplicate_id_ref.as_str())
                        }).collect();

                    if !intersection.is_empty() {
                        Some((idx, intersection))
                    } else {
                        None
                    }
                })
                .collect();
```
a bit longer but in one step